nbconvert/preprocessors/execute.py

```diff
@@ -100,6 +124,9 @@ def run_cell(self, cell):
                 else:
                     raise
 
+            if msg['msg_type'] == 'execute_reply' and msg['metadata']['status'] == 'error' and not self.allow_errors:
+                raise CellExecutionError(msg['content']['traceback'])
```

Comment:
You can put this inside the `if msg['parent_header'].get('msg_id') == msg_id:` check just below, to ensure that we're looking at the message which is a response to this execute request (it shouldn't be anything else, but it's worth checking). And then I think we can lose the msg_type check, because the only message on the shell channel in response to an `execute_request` should be `execute_reply`.